heft-plugins/heft-api-extractor-plugin/src/ApiExtractorPlugin.ts

```diff
@@ -0,0 +1,246 @@
+  ): Promise<void> {
+    const extractorGeneratedFilePaths: string[] = [];
+    if (apiExtractorConfiguration.apiReportEnabled) {
+      // Keep apiExtractorConfiguration.reportFilePath as-is, since API-Extractor uses the existing
+      // content to write a warning if the output has changed.
+      extractorGeneratedFilePaths.push(apiExtractorConfiguration.reportTempFilePath);
+    }
+    if (apiExtractorConfiguration.docModelEnabled) {
+      extractorGeneratedFilePaths.push(apiExtractorConfiguration.apiJsonFilePath);
+    }
+    if (apiExtractorConfiguration.rollupEnabled) {
+      extractorGeneratedFilePaths.push(
+        apiExtractorConfiguration.alphaTrimmedFilePath,
+        apiExtractorConfiguration.betaTrimmedFilePath,
+        apiExtractorConfiguration.publicTrimmedFilePath,
+        apiExtractorConfiguration.untrimmedFilePath
+      );
+    }
+    if (apiExtractorConfiguration.tsdocMetadataEnabled) {
+      extractorGeneratedFilePaths.push(apiExtractorConfiguration.tsdocMetadataFilePath);
+    }
+
+    for (const generatedFilePath of extractorGeneratedFilePaths) {
+      if (generatedFilePath) {
+        cleanOptions.addDeleteOperations({
+          sourceFolder: path.dirname(generatedFilePath),
+          includeGlobs: [path.basename(generatedFilePath)]
+        });
+      }
+    }
+  }
+
+  private async _getApiExtractorTaskConfigurationAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<IApiExtractorPluginConfiguration | undefined> {
+    if (!this._apiExtractorTaskConfigurationPromise) {
+      this._apiExtractorTaskConfigurationPromise = this._getApiExtractorTaskConfigurationInnerAsync(
+        taskSession,
+        heftConfiguration
+      );
+    }
+
+    return await this._apiExtractorTaskConfigurationPromise;
+  }
+
+  private async _getApiExtractorTaskConfigurationInnerAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration
+  ): Promise<IApiExtractorPluginConfiguration | undefined> {
+    const apiExtractorTaskConfigurationFileLoader: ConfigurationFile<IApiExtractorPluginConfiguration> =
+      new ConfigurationFile<IApiExtractorPluginConfiguration>({
+        projectRelativeFilePath: 'config/api-extractor-task.json',
+        jsonSchemaPath: PLUGIN_SCHEMA_PATH
+      });
+
+    return await apiExtractorTaskConfigurationFileLoader.tryLoadConfigurationFileForProjectAsync(
+      taskSession.logger.terminal,
+      heftConfiguration.buildFolder,
+      heftConfiguration.rigConfig
+    );
+  }
+
+  private async _runApiExtractorAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    runOptions: IHeftTaskRunHookOptions,
+    apiExtractor: typeof TApiExtractor,
+    apiExtractorConfiguration: TApiExtractor.ExtractorConfig
+  ): Promise<void> {
+    // TODO: Handle watch mode
+    // if (watchMode) {
+    //   taskSession.logger.terminal.writeWarningLine("API Extractor isn't currently supported in --watch mode.");
```

Comment:
It'd be cool if it actually was supported.

Reply:
Agree. I added the TODO specifically so I can go back and support